httpd/handler.go

```diff
@@ -376,6 +380,29 @@ func authenticate(inner func(http.ResponseWriter, *http.Request, *influxdb.User)
 	})
 }
 
+type gzipResponseWriter struct {
+	io.Writer
+	http.ResponseWriter
+}
+
+func (w gzipResponseWriter) Write(b []byte) (int, error) {
+	return w.Writer.Write(b)
+}
+
+// determines if the client can accept compressed responses, and encodes accordingly
+func gzipFilter(inner http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") {
+			inner.ServeHTTP(w, r)
```

Comment:
To be clear about this, `inner.ServerHTTP` will be called if `gzip` is not supported, then the `else` will be supported. Should it instead return?